src/engine/QueryPlanner.cpp

```diff
@@ -973,14 +1020,27 @@ Variable QueryPlanner::generateUniqueVarName() {
 // _____________________________________________________________________________
 QueryPlanner::SubtreePlan QueryPlanner::getTextLeafPlan(
     const QueryPlanner::TripleGraph::Node& node) const {
+  AD_CONTRACT_CHECK(node._wordPart.has_value());
+  string word = node._wordPart.value();
   SubtreePlan plan(_qec);
+  if (node._triple._p._iri == CONTAINS_ENTITY_PREDICATE) {
+    if (node._variables.size() == 2) {
+      Variable evar = *(node._variables.begin()) == node._cvar.value()
+                          ? *(++node._variables.begin())
+                          : *(node._variables.begin());
```

Comment:
Add a `TODO<joka921>`:This is not nice, refactor the whole TripleGraph class to make these checks more explicity.